python/ray/remote_function.py

```diff
@@ -217,6 +218,14 @@ def _remote(self, args=None, kwargs=None, **task_options):
 
         # fill task required options
         for k, v in ray_option_utils.task_options.items():
+            if k == "max_retries":
+                # TODO(swang): We need to override max_retries here because the default
```

Comment:
ah this is tricky... wondering if there is a better way.

Reply:
Just curious, but Is this not going to work?
```
    "max_retries": _counting_option(
        "max_retries", default_value=os.environ.get("RAY_TASK_MAX_RETRIES", ray_constants.DEFAULT_TASK_MAX_RETRIES)
    ),
```
?

Reply:
It doesn't work if the OS env is set after the ray import. Maybe it's only a real issue for python tests, but either way checking it at ray.remote time seems more robust.